### ISSUE TITLE
One more issue about the nondecreasing level of loss function

### ISSUE THREAD

**Issue by medphisiker** [qubvel/segmentation_models#105]
Hello, 
I am trying to create a neural network **Unet** by segmentation_models. I have a few questions.
According to the documentation, segmentation models are standard models of Keras and possess their methods. The documentation provides an example of training model by **model.fit()** Instead, I tried to use **model.fit_generator ()** for this purpose. Is it possible ? My code doesn’t work.
Then I am try to train **Unet** by standard documentation way by **model.fit().**
In both cases, It doesn't work. Losses function values are constants.
According to the issue (https://github.com/qubvel/segmentation_models/issues/74 or https://github.com/qubvel/segmentation_models/issues/93) the input image must be processed using the **preprocess_input()** function that we imported above.
Make sure that during training / validation / test / inference stages:
**loaded images uint8 dtype, range 0-255, dtype 'uint8'
loaded masks have values {0, 1}, dtype 'uint8'**
Check that before your x is uint8 data (0..255 range) both during training and during testing.
I checked my data and they **meet the conditions described above.**
Then I create Unet manually by Keras, and it works nice with validation binary accuracy about 99 %.
Please, help me figure out what I'm doing wrong.
My python Jupiter notebook attached on link below:
https://drive.google.com/open?id=1Bb1XuZIZ5etwzOOtMJe7MtIDoKKoAo9P

**Comment by qubvel** [comment]
Hi @medphisiker 
For binary segmentation (you have masks with shape (16, 256, 256, **1**)) net `activation` should be `sigmoid` not `softmax`.

**Comment by medphisiker** [comment]
I investigated the code and the error was in logic ! I'm embarrassed for this mistake.
Thank you so much for the quick response and the hint. 
For now it's works nice with **model.fit()** and **model.fit_generator ()**.
As well thank for your wonderful library ! )
Perhaps, it may be interesting to add an example with **model.fit_generator (**) to documentation. When I read the documentation for the first time I decided that the library only works with **model.fit()**.
